File: BusReservation/ClientApp/src/app/shared/Model/Ticket.ts
import { BusRoute } from './BusRoute';
import { Discount } from './Discount';
import { User } from './User';

export class Ticket {
    ticketId: number;
    ticketNumber: number;
    travelDate: Date;
    seat: number;
    discount: Discount;
    busRoute: BusRoute;
    user: User;
    creationDate: Date;
    deletionDate: Date;
}

File: BusReservation.Model/Ticket.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace BusReservation.Model
{
    public class Ticket
    {
        public int TicketId { get; set; }
        public int TicketNumber { get; set; }
        public DateTime TravelDate { get; set; }
        public int Seat { get; set; }
        public Discount Discount { get; set; }
        public BusRoute BusRoute { get; set; }
        public User User { get; set; }
        public DateTime CreationDate { get; set; }
        public DateTime DeletionDate { get; set; }
        public bool IsBought { get; set; }
    }
}
